(DOCSP-19703) Index Page (#2)
mongodb/docs-visual-studio-extension: source/index.txt

.. _visual-studio-extension:

.. _visual-studio-extension-landing:

===============================
{+product+}
===============================

.. default-domain:: mongodb

.. toctree::
   :titlesonly:
   :maxdepth: 1

   Installation </install>
   Builders </builders>
   LINQ </linq>
   Configuration </configuration>
   FAQ </faq>
   What's New </whats-new>

Overview
--------

> The `{+product+} <{+product-home-page+}>`__ is a NuGet package

Reviewer: Maybe something like: "is a Roslyn analyzer that..." or "is a Roslyn analyzer (packaged as a NuGet package) that..."

Author: I agree "NuGet package" should be swapped. I'm nervous mentioning "Roslyn Analyzer" in the introductory sentence of the guide as I think it may make readers think they need to understand "Roslyn" to use the MongoDB Analyzer (I'm not sure they need to understand the Roslyn compiler to use the MongoDB Analyzer but I could be mistaken).

Would you be comfortable replacing "NuGet package" with the word "tool"? Based off the [Sarif specification](https://github.com/microsoft/sarif-tutorials/blob/main/docs/1-Introduction.md#about-static-analysis-tools), it seems like it is appropriate to refer to a static analysis tool as a "tool":

> We often refer to a static analysis tool simply as a "tool."

If you think "Roslyn Analyzer" is a better term than "tool" however I'll use "Roslyn Analyzer". Just wanted to raise my concern about introducing "Roslyn" too soon.

Author: I'll use "Roslyn Analyzer" for the time being in order to merge this PR so that I can get the next page into review. When you have a free moment, let me know your thoughts on the term "tool" and if you like the switch I'll add make the change as part of a different PR.

Author: Discussed on slack, and we agreed "tool" is acceptable here.

that helps you understand the following about your {+driver-long+} 
code:

- How your code translates into the {+query-api+}

> - Does your code include an unsupported LINQ or builder expression

Reviewer 2: ```suggestion
- If your code includes unsupported LINQ or builder expressions
```

Author: Took suggestion!

The {+product+} can analyze the following groups of {+driver-short+} expressions:

- `Builders <{+driver-docs+}reference/driver/definitions/>`__
- `LINQ <{+driver-docs+}reference/driver/crud/linq/>`__
- `LINQ3 <{+driver-docs+}reference/driver/crud/linq3/>`__

Read the following sections of this guide to learn how to install the
{+product+}, how to use its features, and how to configure it:

- :doc:`Install the extension </install>`
- :doc:`Analyze your builder expressions </builders>`
- :doc:`Analyze your LINQ expressions </linq>`
- :doc:`Configure the extension </configuration>`

To view answers to frequently asked questions, see the :doc:`FAQ </faq>` page.

To view the differences between versions of the extension,
see the :doc:`What's New <whats-new>` page. 

> .. note:: Roslyn Analyzer

Reviewer: Nice, I thought to add this to FAQ, but "note" actually looks better.

Author: Thanks! And we can do both if you think it makes sense to add to the FAQ as well. I'll link the preceding comment to the [FAQ page JIRA ticket](https://jira.mongodb.org/browse/DOCSP-19710) so we remember to consider Roslyn Analyzers when writing the FAQ page.

.. _visual-studio-extension:

.. _visual-studio-extension-landing:

===============================
{+product+}
===============================

.. default-domain:: mongodb

.. toctree::
   :titlesonly:
   :maxdepth: 1

   Installation </install>
   Builders </builders>
   LINQ </linq>
   Configuration </configuration>
   FAQ </faq>
   What's New </whats-new>

Overview
--------

> The `{+product+} <{+product-home-page+}>`__ is an add-on for the Visual Studio IDE
> that uses **static analysis** to help you understand your {+driver-long+} 
> code. Static analysis is a technique where software inspects your
> code without running it.

Reviewer 3: issue:  Since this isn't a "guide" like we normally author on a fundamentals page, I'm not sure this approach is the best.

What we should be doing is putting the value proposition up front and lettings users decide if they want to use the product and keep reading further.

We can use a lot of the information from the product scope document for this intro. See the heading for "How it Works" to confirm.
```suggestion
The `{+product+} <{+product-home-page+}>`__ is an add-on for the Visual Studio IDE
that displays generated {+query-api+}, flags invalid and unsupported LINQ expressions, and more to aid in application development with MongoDB.
```

Users don't need to know what static analysis is to understand the benefit of the product or how to use the product, so I think including the term and the definition isn't beneficial.

Author: I removed the mention of static analysis and moved the things the extension can do immediately after the first sentence to quickly provide readers with a value proposition. Let me know if this solves the issues!
